lib/npg_tracking/glossary/moniker.pm

```diff
@@ -35,6 +38,45 @@ sub file_name {
   return $name;
 }
 
+sub file_name_full {
+  my ($self, $name, @options) = @_;
+
+  $name or croak 'File name base should be given';
+
+  my %h;
+  if (@options) {
+    try {
+      use warnings FATAL => qw/misc/; # Turn warning
+                                      # 'Odd number of elements in hash assignment'
+                                      # into an error...
+      %h = @options;
+    } catch {
+      my $e = $_;
+      if ($e =~ /Odd\ number\ of\ elements\ in\ hash\ assignment/smxi) {
+        $e = 'The argument list should contain a file name'.
+             ' and, optionally, a list of key-value options';
+      }
+      croak $e;
+    };
+    if (exists $h{$SUFFIX_KEY} && $h{$SUFFIX_KEY} ne q[]) {
+      $name .= q[_] . $h{$SUFFIX_KEY};
+      delete $h{$SUFFIX_KEY};
+    }
+    if (exists $h{$EXT_KEY} && $h{$EXT_KEY} ne q[]) {
+      $name .= q[.] . $h{$EXT_KEY};
+      delete  $h{$EXT_KEY};
+    }
+    my @other = keys %h;
+    if (@other) {
```

Comment:
No need to create @other; just check on number of keys in %h, and sort its keys on line 72.